lib/job_helpers/email_job_helper.rb

```diff
@@ -0,0 +1,19 @@
+module EmailJobHelper
+  private
+
+  def send_email(res)
+    UserMailer.reservation_status_update(res).deliver_now
+  end
+
+  # queries can just be a list of symbols like :checked_out, :not_overdue
+  # but if you want to send arguments to the query method, wrap them in an array
+  # like [:not_flagged, :missed_email_sent]
+  def collection(*queries) 
+    @set ||= queries.inject(Reservation) { |result, query| result.send(*query) }
+  end
+
+  def clean
+    @set = nil
+  end
```

Comment:
if we decide that we like this setup, I'll move the collection related methods to a more generic helper